modules/custom/falcon_thankq/src/Plugin/Field/FieldWidget/InlineSourceCodeForm.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\falcon_thankq\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element;
+use Drupal\inline_entity_form\Plugin\Field\FieldWidget\InlineEntityFormComplex;
+
+/**
+ * Source codes field widget.
+ *
+ * @FieldWidget(
+ *   id = "inline_source_code_form",
+ *   label = @Translation("Inline Source Codes form"),
+ *   field_types = {
+ *     "entity_reference"
+ *   },
+ *   multiple_values = true
+ * )
+ */
+class InlineSourceCodeForm extends InlineEntityFormComplex {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+
+    // Mainly use parent code to render the form.
+    $element = parent::formElement($items, $delta, $element, $form, $form_state);
+
+    // Change field label to the one related to source codes.
+    $element['entities']['#table_fields']['label']['label'] = t('Source code');
+
+    foreach (Element::children($element['entities']) as $key) {
+
+      // Get Source Code entity.
+      /* @var $entity \Drupal\eck\EckEntityInterface */
+      $entity = $element['entities'][$key]['#entity'];
+
+      // Grab source code & human readable label.
+      $source_code = $entity->get('field_source_code')->getString();
+      $label = $entity->get('field_label')->getString();
```

Comment:
Looks unusual to me... Usually it is enough to write smth like `$entity->field_label->value`.

Reply:
@duozersk I don't have a strong preference, happy with your version cause it's more minimalistic.